Sources/CoreMetrics/SystemMetrics.swift

```diff
@@ -0,0 +1,332 @@
+            self.interval = interval
+            if let dataProvider = dataProvider {
+                self.dataProvider = dataProvider
+            } else {
+                #if os(Linux)
+                self.dataProvider = SystemMetrics.linuxSystemMetrics
+                #else
+                self.dataProvider = SystemMetrics.noopSystemMetrics
+                #endif
+            }
+            self.labels = labels
+        }
+    }
+
+    /// Labels for the reported System Metrics Data.
+    ///
+    /// Backend implementations are encouraged to provide a static extension with
+    /// defaults that suit their specific backend needs.
+    public struct Labels {
+        /// Prefix to prefix all other labels with.
+        let prefix: String
+        /// Virtual memory size in bytes.
+        let virtualMemoryBytes: String
+        /// Resident memory size in bytes.
+        let residentMemoryBytes: String
+        /// Total user and system CPU time spent in seconds.
+        let startTimeSeconds: String
+        /// Total user and system CPU time spent in seconds.
+        let cpuSecondsTotal: String
+        /// Maximum number of open file descriptors.
+        let maxFileDescriptors: String
+        /// Number of open file descriptors.
+        let openFileDescriptors: String
+
+        /// Create a new `Labels` instance.
+        ///
+        /// - parameters:
+        ///     - prefix: Prefix to prefix all other labels with.
+        ///     - virtualMemoryBytes: Virtual memory size in bytes
+        ///     - residentMemoryBytes: Resident memory size in bytes.
+        ///     - startTimeSeconds: Total user and system CPU time spent in seconds.
+        ///     - cpuSecondsTotal: Total user and system CPU time spent in seconds.
+        ///     - maxFds: Maximum number of open file descriptors.
+        ///     - openFds: Number of open file descriptors.
+        public init(prefix: String, virtualMemoryBytes: String, residentMemoryBytes: String, startTimeSeconds: String, cpuSecondsTotal: String, maxFds: String, openFds: String) {
+            self.prefix = prefix
+            self.virtualMemoryBytes = virtualMemoryBytes
+            self.residentMemoryBytes = residentMemoryBytes
+            self.startTimeSeconds = startTimeSeconds
+            self.cpuSecondsTotal = cpuSecondsTotal
+            self.maxFileDescriptors = maxFds
+            self.openFileDescriptors = openFds
+        }
+
+        func label(for keyPath: KeyPath<Labels, String>) -> String {
+            return self.prefix + self[keyPath: keyPath]
+        }
+    }
+
+    /// System Metric data.
+    ///
+    /// The current list of metrics exposed is taken from the Prometheus Client Library Guidelines
+    /// https://prometheus.io/docs/instrumenting/writing_clientlibs/#standard-and-runtime-collectors
+    public struct Data {
```

Comment:
This name is not very expressive. Maybe `ProcessStatistics` would be a better name?

Reply:
Keep in mind the full name would be `SystemMetrics.Data` since it's a type inside of a different type. Having `SystemMetrics.ProcessStatistics` might become a bit too verbose, if that makes sense 😄 

Agree that `Data` could be improved though 👍